cloudify/utils.py

```diff
@@ -1058,3 +1062,23 @@ def uuid4():
         uuid_as_hex[16:20],
         uuid_as_hex[20:]
     )
+
+
+def keep_trying_http(total_timeout_sec=KEEP_TRYING_HTTP_TOTAL_TIMEOUT_SEC,
+                     max_delay_sec=MAX_WAIT_BETWEEN_HTTP_RETRIES_SEC):
+    logger = setup_logger('http_retrying')
+
+    def wrapper(func):
+        timeout_at = None if total_timeout_sec is None \
+            else datetime.utcnow() + timedelta(seconds=total_timeout_sec)
+        while True:
+            try:
+                return func
+            except (ConnectionError, Timeout):
+                if timeout_at and datetime.utcnow() > timeout_at:
+                    raise
+            delay = random.randint(1, max_delay_sec)
+            logger.info(f'Will retry {func} in {delay} seconds')
```

Comment:
This should probably include the error so that you can see what it is (and maybe be able to fix it or figure out why it happened)

Reply:
Oh, right.  I was expecting client to produce some logs but as it turns out, it's the case only if debugging is enabled…  Thanks.